### PR TITLE
[embedded] Adjust docs to clarify support for class-bound existentials

### PR DESCRIPTION
**Pull request by kubamracek**

### CODE REVIEW

**Review by rauhul — APPROVED** [review]

**Comment by kubamracek** [comment]
@swift-ci please smoke test

File: docs/EmbeddedSwift/UserManual.md [hunk 1]
```diff
@@ -183,7 +183,7 @@ Embedded Swift is a subset of the Swift language, and some features are not avai
 Features that are not available:
 
 - **Not available**: Runtime reflection (`Mirror` APIs).
-- **Not available**: Values of protocol types ("existentials"), e.g. `let a: Hashable = ...`, are not allowed. `Any` and `AnyObject` are also not allowed.
+- **Not available**: Values of protocol types ("existentials"), unless the protocol is restricted to be class-bound (derived from AnyObject). E.g. `let a: Hashable = ...` is not allowed. `Any` is also not allowed.
```
Comment (slavapestov) [inline docs/EmbeddedSwift/UserManual.md:186]: Is the restriction really on the _protocol_, or just the existential type? That is, do we allow `any Hashable & AnyObject`, `any Hashable & SomeClassBoundProtocol`, or `any Hashable & SomeClass`?

Comment (eeckstein) [reply docs/EmbeddedSwift/UserManual.md:186]: It is really on the protocol - at least currently.

Comment (slavapestov) [reply docs/EmbeddedSwift/UserManual.md:186]: Why?

**Pull request merged**